app/dao/notifications_dao.py

```diff
@@ -438,17 +438,32 @@ def move_notifications_to_notification_history(
     )
 
 
-def delete_test_notifications(notification_type, service_id, timestamp_to_delete_backwards_from):
+def delete_test_notifications(notification_type, service_id, timestamp_to_delete_backwards_from, qry_limit=50000):
+    if notification_type == LETTER_TYPE:
+        # reduced query limit so we don't run into issues trying to loop through 50k letters in python deleting from s3
+        # use `min` so we can reduce query limit artificially during unit tests
+        qry_limit = min(qry_limit, 5_000)
+        _delete_test_letters_from_s3(service_id, timestamp_to_delete_backwards_from, qry_limit)
+
     # Deleting test Notifications, test notifications are not persisted to NotificationHistory
-    Notification.query.filter(
-        Notification.notification_type == notification_type,
-        Notification.service_id == service_id,
-        Notification.created_at < timestamp_to_delete_backwards_from,
-        Notification.key_type == KEY_TYPE_TEST,
-    ).delete(synchronize_session=False)
+    subquery = (
+        db.session.query(Notification.id)
+        .filter(
+            Notification.notification_type == notification_type,
+            Notification.service_id == service_id,
+            Notification.created_at < timestamp_to_delete_backwards_from,
```

Comment:
Could we have some sort of limit of earliest notification to consider? Otherwise we're probably doing the whole sequential-scan-of-the-table-back-to-the-beginning again...

Reply:
Actually come to think, it might not be such a big deal given it's restricted to a single service id